apps/mobile/src/features/browser/approver-sheet/stx/sign-transaction/sign-transaction.tsx

```diff
@@ -0,0 +1,88 @@
+import { useState } from 'react';
+
+import { getTxOptions } from '@/features/approver/utils';
+import { useAccounts } from '@/store/accounts/accounts.read';
+import { App } from '@/store/apps/utils';
+import { useStacksSigners } from '@/store/keychains/stacks/stacks-keychains.read';
+import {
+  getStacksNetworkFromName,
+  useNetworkPreferenceStacksNetwork,
+} from '@/store/settings/settings.read';
+import { destructAccountIdentifier } from '@/store/utils';
+import { StacksNetworkName, StacksNetworks } from '@stacks/network';
+import { deserializeTransaction } from '@stacks/transactions';
+
+import {
+  RpcRequest,
+  RpcResponse,
+  createRpcSuccessResponse,
+  stxSignTransaction,
+} from '@leather.io/rpc';
+
+import { SignTransactionApproverLayout } from './sign-transaction.layout';
+
+interface SignTransactionApproverProps {
+  app: App;
+  message: RpcRequest<typeof stxSignTransaction>;
+  sendResult(result: RpcResponse<typeof stxSignTransaction>): void;
+  origin: string;
+  closeApprover(): void;
+}
+
+export function SignTransactionApprover(props: SignTransactionApproverProps) {
+  const network = useNetworkPreferenceStacksNetwork();
+  function getNetwork() {
+    if (!props.message.params.network) {
+      return network;
+    }
+    if (StacksNetworks.includes(props.message.params.network as StacksNetworkName)) {
```

Comment:
Can we:
- clean `props.message.params.network` up with a destructure? 
- avoid casting `as StacksNetworkName)`
- add copy for the Errors? - will the users see those?

Reply:
Agree with all but Pete's [overzealous destructuring](https://www.aleksandrhovhannisyan.com/blog/overzealous-destructuring/)

Reply:
If we don't want to either read from deeply nested, or use deep destructuring, I'd suggest selector or lens pattern

Reply:
On type casting https://trustmachines.slack.com/archives/C05LAP952E7/p1743775904159909